aptos-move/block-executor/src/executor.rs

```diff
@@ -1091,20 +1087,14 @@ where
                         &shared_commit_state,
                         &final_results,
                     ) {
-                        if scheduler.halt() {
-                            // Only one thread / worker will successfully halt, hence below
-                            // ExplicitSyncWrapper acquires are safe.
-
-                            if let BlockExecutionError::FatalVMError((inner_err, txn_idx)) = &e {
-                                let block_limit_processor = shared_commit_state.acquire();
-                                info!("Block execution was aborted due to {:?}", inner_err);
-                                block_limit_processor.finish_parallel_update_counters_and_log_info(
-                                    txn_idx + 1,
-                                    scheduler.num_txns(),
-                                );
-                            }
-
-                            *shared_maybe_error.acquire() = Err(e);
+                        log_panic_or(&e, "[BlockSTM] Parallel execution worker loop halt_err");
```

Comment:
Why does log come before `if scheduler.halt_err()`?

Reply:
because we want to log all errors, but only one of them will trigger the fallback

Reply:
(I wonder if I should add transaction index everywhere, that might be helpful)